sql/sql_statistics.h

```diff
@@ -476,6 +476,7 @@ class Index_statistics
 
   void set_avg_frequency(uint i, double val)
   {
+    if (isnan(val)) return;
     avg_frequency[i]= (ulonglong) (val * Scale_factor_avg_frequency);
```

Comment:
`/server/sql/sql_statistics.h:479:35: runtime error: -nan is outside the range of representable values of type 'unsigned long long'`